src/components/wyehome/OutOfOfficeEdit.jsx

```diff
@@ -0,0 +1,254 @@
+        this.setState({ toDate: value });
+        document.getElementById('toDate').value = value;
+      }
+    }
+  };
+
+  handleDelete = () => {
+    const { reason, fromDate, toDate } = this.state;
+    const { idOoo } = this.props;
+    const userId = parseInt(window.localStorage.getItem('sessionUserId'), 10);
+    const endPoint = `${process.env.REACT_APP_API_URL}/out_of_office/${idOoo}`;
+    fetch(endPoint, {
+      method: 'DELETE',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${window.localStorage.getItem('sessionToken')}`,
+      }),
+      body: JSON.stringify({
+        out_office_type: reason,
+        begin_at: `${fromDate}T00:00:00`,
+        finish_at: `${toDate}T23:59:59`,
+        user_id: userId,
+      }),
+    })
+      .then((response) => {
+        if (response.ok) {
+          const { onModeChange, showAlert } = this.props;
+          showAlert('Out of Office was deleted.', 'info');
+          onModeChange('list');
+          return null;
+        }
+        return response.json();
+      }).then((json) => {
+        if (json !== null) {
+          const { onModeChange, showAlert } = this.props;
+          showAlert('An error occured while deleting Out of Office. Try again later.', 'warning');
+          onModeChange('list');
+        }
+      }).catch(() => {
+        const { onModeChange, showAlert } = this.props;
+        showAlert('An error occured while deleting Out of Office. Try again later.', 'warning');
+        onModeChange('list');
+      });
+  };
+
+
+  handleSubmit = (e) => {
+    const {
+      reason, fromDate, toDate,
+    } = this.state;
+    const { idOoo } = this.props;
+    e.preventDefault();
+    const endPoint = `${process.env.REACT_APP_API_URL}/out_of_office/${idOoo}`;
+    const userId = parseInt(window.localStorage.getItem('sessionUserId'), 10);
+    fetch(endPoint, {
+      method: 'PUT',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${window.localStorage.getItem('sessionToken')}`,
+      }),
+      body: JSON.stringify({
+        out_office_type: reason.value,
+        begin_at: `${fromDate}T00:00:00`,
+        finish_at: `${toDate}T23:59:59`,
+        user_id: userId,
+      }),
+    })
+      .then((response) => {
+        if (response.ok) {
+          const { onModeChange, showAlert } = this.props;
+          showAlert('Out of Office was updated!', 'info');
+          onModeChange('list');
+          return null;
+        }
+        return response.json();
+      }).then((json) => {
+        if (json !== null) {
+          const { showAlert } = this.props;
+          showAlert(json[0], 'warning');
+        }
+      }).catch(() => {
+        const { onModeChange, showAlert } = this.props;
+        showAlert('An error occured while updating Out of Office. Try again later.', 'warning');
+        onModeChange('list');
+      });
+  };
+
+  toggle() {
```

Comment:
Maybe use a more descriptive name? Like:
```suggestion
  toggleModal() {
```